ruby-challenges.rb

```diff
@@ -10,23 +10,108 @@
+def hash_it_up_bro (hash)
+    hash.values.flatten.sort
+end
 
-# Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
+p hash_it_up_bro(us_states)
+# ["Arizona", "California", "Idaho", "Maine", "Nevada", "New Hampshire", "Oregon", "Rhode Island", "Washington"]
 
+# --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
+# Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
 # -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed by a given amount. The brake method should decrease the speed by a given amount. The bike cannot go negative speeds.
 
 # Expected output example: my_bike.pedal_faster(10) => 10
 # Expected output example: my_bike.pedal_faster(18) => 28
 # Expected output example: my_bike.brake(5) => 23
 # Expected output example: my_bike.brake(25) => 0
+
+# pseudo:
+# create a class called Bike
+# define an initialize method that takes in an arguement for the model. We do not need to pass an argument for wheels or current_speed as they have default values. In the future we can add methods to change the wheels value, but it is outide of the scope for this challenge.
+# create an attr_accessor for :model, :wheels, and :current_speed so we can update the attributes that we have initialized.
+# create another method in our class called bike_info that returns a string with string interpolation describing the model of the bike, how many wheels it has, and its current speed.
+# define a method called pedal_faster that takes in and integer as an argument (speed) that when called on reassigns @current_speed to @current_speed + speed. return a string with string interpolation stating the current speed of the bike.
+# finally, define a method called brake that takes in an integer as an argument (brake_power) that when called on reassigns @current_speed to @current_speed - speed. return a string with string interpolation stating the current speed of the bike. within the brake method create a conditional statement that checks if @current_speed less than or equal to 0, if that is true, set @current_speed to 0 and return "You have come to a stop."
+
+class Bike 
+    attr_accessor :model, :wheels, :current_speed
+    def initialize(model)
+        @model = model
+        @wheels = 2
+        @current_speed = 0
+    end
+    def bike_info
+        "The #{@model} bike has #{@wheels} wheels and is going #{@current_speed} mph."
+    end
+    def pedal_faster(speed)
+        @current_speed = @current_speed + speed
```

Comment:
You crushed this section! You can use the += operator if you don't want to type @current_speed twice but other than that, looks great! Are you sure you don't like Ruby as much as JS?